examples/ansible/setup_kea_dhcp_server.yml

```diff
@@ -0,0 +1,87 @@
+#   - 'dhcp4_template_src': Path to the DHCPv4 configuration j2 template. Leave empty if not using a template.
+#   - 'dhcp6_template_src': Path to the DHCPv6 configuration j2 template. Leave empty if not using a template.
+#   - 'kea_config_path': Path where the Kea DHCP server configuration files will be stored.
+#   - 'workload': A dictionary containing information about the Kea workload, including its name and container image.
+
+- name: Deploying and Managing the Kea DHCP server workload
+  hosts: alphost
+  become: true
+  vars:
+    run_dhcpv6: false
+    dhcp4_template_src: ""
+    dhcp6_template_src: ""
+    kea_config_path: /etc/kea
+    workload:
+      name: kea
+      image: registry.opensuse.org/suse/alp/workloads/tumbleweed_containerfiles/suse/alp/workloads/kea:latest
+
+  tasks:
+    - name: Setup Kea DHCP server
+      block:
+        - name: Pull the Kea DHCP server container image
+          containers.podman.podman_image:
+            name: "{{ workload.image }}"
+            state: present
+
+        - name: Install all required parts of the Kea workload
+          ansible.builtin.command: >-
+            podman container runlabel install {{ workload.image }}
+          register: workload_runlabel_install
+          changed_when:
+            - ('already exist' not in workload_runlabel_install.stdout)
+
+        - name: Add firewall exception rule for DHCPv6
+          ansible.posix.firewalld:
+            service: dhcpv6
+            permanent: yes
+            state: enabled
+            immediate: yes
+          when:
+            - run_dhcpv6
+
+        - name: Generate Kea DHCPv4 configuration from template
+          ansible.builtin.template:
+            src: "{{ dhcp4_template_src }}"
+            dest: "{{ kea_config_path }}/kea-dhcp4.conf"
+          register: dhcp4_config
+          notify: Reload Kea configuration
+          when:
+            - dhcp4_template_src != ""
+
+        - name: Generate Kea DHCPv6 configuration from template
+          ansible.builtin.template:
+            src: "{{ dhcp6_template_src }}"
+            dest: "{{ kea_config_path }}/kea-dhcp6.conf"
+          register: dhcp6_config
+          notify: Reload Kea configuration
+          when:
+            - dhcp6_template_src != ""
+
+        - name: Start Kea DHCPv4 server using systemd
+          ansible.builtin.systemd:
+            name: kea-dhcp4.service
+            state: started
+            enabled: yes
+          when:
+            - not run_dhcpv6
+
+        - name: Start Kea DHCPv6 server using systemd
+          ansible.builtin.systemd:
+            name: kea-dhcp6.service
+            state: started
+            enabled: yes
+          when:
+            - run_dhcpv6
+
+  handlers:
+    - name: Reload Kea configuration
+      ansible.builtin.command: /usr/local/bin/keactrl reload
+      when:
+        - dhcp4_config.changed or dhcp6_config.changed
```

Comment:
This is not needed. It's already handled internally in Ansible in the `notify` section.

Reply:
Hi Pavel, I will make the above changes and split the playbook into two so that it is easier to manage the config vars for the dhcpv4 and dhcpv6 templates.